services/src/pro/datasets/external/sentinel_s2_l2a_cogs.rs

```diff
@@ -466,6 +466,35 @@ impl SentinelS2L2aCogsMetaData {
                 query.time_interval.end() + query_end_buffer
             };
 
+            // clip end to the same day to avoid filling wrong days
```

Comment:
not wrong days, but partial days right? Please explain it some more. Does it have to do with our aggregation granularity? Otherwise I am not sure why it would matter.

Reply:
https://github.com/geo-engine/geoengine/pull/989/commits/02c4d8e7f32d2916e6c038d9e60b1751752d0fd9